programs/mango-v4/src/state/token_conditional_swap.rs

```diff
@@ -148,10 +194,22 @@ impl TokenConditionalSwap {
         self.has_data = u8::from(has_data);
     }
 
+    pub fn tcs_type(&self) -> TokenConditionalSwapType {
+        self.tcs_type.try_into().unwrap()
+    }
+
     pub fn is_expired(&self, now_ts: u64) -> bool {
         now_ts >= self.expiry_timestamp
     }
 
+    pub fn passed_start(&self, now_ts: u64) -> bool {
+        self.start_timestamp > 0 && now_ts >= self.start_timestamp
+    }
+
+    pub fn has_incentive_for_starting(&self) -> bool {
+        self.tcs_type() == TokenConditionalSwapType::PremiumAuction
```

Comment:
hmm so the linear auction has no incentive to start?

Reply:
Linear auctions always start at a pre-defined start_timestamp. There's no need to call start() once a condition is met.